src/clj/web/app_state.clj

```diff
@@ -4,11 +4,13 @@
 
 (defonce app-state
   (atom {:lobbies {}
+         :lobby-updates {}
          :users {}}))
 
 (defn register-user
   [app-state uid user]
-  (assoc-in app-state [:users uid] (assoc user :uid uid :lobby-updates true)))
+  (merge app-state {:users (assoc (:users app-state) uid (assoc user :uid uid))
```

Comment:
Instead of merge, let's do a thread-first and two `assoc-in`s, as `merge` isn't a deep merge.

Reply:
👍🏻